components/script/dom/blob.rs

```diff
@@ -88,22 +90,76 @@ impl DataSlice {
 pub struct Blob {
     reflector_: Reflector,
     #[ignore_heap_size_of = "No clear owner"]
-    data: DataSlice,
+    blob_impl: BlobImpl,
     typeString: String,
     isClosed_: Cell<bool>,
 }
 
+#[derive(Clone, JSTraceable)]
+pub struct BlobImpl {
+    slice: DOMRefCell<Option<DataSlice>>,
```

Comment:
I'd rather have an enum here and within the FileId variant have an option for caching. As it stands slice = Some can mean cached or an in-memory slice, a clear boundary would be nicer and more Rusty.